packages/mobile/src/app/WebViewScreen.tsx

```diff
@@ -37,29 +71,66 @@ function WebViewScreen({ route }: Props) {
     return true
   }
 
-  const hideLoading = () => {
-    setLoading(false)
+  const handleRefresh = () => {
+    if (webviewRef && webviewRef.current) {
+      webviewRef.current.reload()
+    }
+  }
+
+  const handleGoForward = () => {
+    if (webviewRef && webviewRef.current) {
+      webviewRef.current.goForward()
+    }
+  }
+
+  const handleGoBack = () => {
+    if (webviewRef && webviewRef.current) {
+      webviewRef.current.goBack()
+    }
   }
 
   return (
-    <View style={styles.container}>
+    <SafeAreaView style={styles.container} edges={['bottom']}>
       <WebView
+        ref={webviewRef}
         originWhitelist={['https://*', 'celo://*']}
         onShouldStartLoadWithRequest={handleLoadRequest}
         setSupportMultipleWindows={false}
         source={{ uri }}
-        onLoad={hideLoading}
+        startInLoadingState={true}
+        renderLoading={() => <ActivityIndicator style={styles.loading} size="large" />}
+        onNavigationStateChange={(navState) => {
+          setCanGoBack(navState.canGoBack)
+          setCanGoForward(navState.canGoForward)
+        }}
       />
-      {loading && <ActivityIndicator style={styles.loading} size="large" />}
-    </View>
+      <View style={styles.navBar}>
```

Comment:
I think this is expected only for iOS according to the designs.
Is that something we can do? I mean, add some code only for iOS or Android.

Reply:
Yes! You can use the `Platform.OS` constant that can be either `android` or `ios`

Reply:
However, there would be no forward button in Android in that case, so I'm not sure that's what we want, even if we said that in the previous meeting 🤔

Reply:
Also for @nityarao, I have to say that I'm bit concern about the UX.
We would have the refresh button at the top while the arrows for navigation at the bottom. I feel it's a bit weird.
I think it would nice to have all those buttons together.
Also, for Android users, what is the expected behaviour if they click on the back button on their devices? Should they go back to the previous webApp page, or should they return to the previous Valora page (e.g. dapps marketplace or homescreen)

Reply:
@gnardini Thats a good catch about not having a forward button in Android. I'd think that users would benefit from navigating both forward & back on dapp pages. 

@dievazqu that being said, what do you think about having the same bottom bar for both iOS & Android to start? 

The icons would all be grouped together in the bottom nav bar. This was my original idea but concerns were raised around it being repetitive with the device back button. In this case, both the device back button & browser back button can take the users back one screen. The close button at the top will close the browser.

Reply:
thanks @dievazqu for bringing this up! grateful for a diverse set of users amongst us 🙏🏻 indeed i didn't think too much on the layout for android before, but now I am thinking:

- while it's nice to be able to browse back and forth, it seems that this is not what Android browsers (like Chrome) easily enable. if this is not part of the "usual" browsing experience for android users, having the bottom nav there might not be useful for them (might be confusing, and provide functionality they don't use). i lean towards Diego's idea of hiding the bottom nav for Android users, to give a more native experience. @nityarao  if you see in this screenshot, it does look very bizarre to have the bottom bar + the hardware buttons and takes some screen real estate away 
<img width="415" alt="Screenshot 2022-03-15 at 09 41 26" src="https://user-images.githubusercontent.com/20150449/158340054-29c847e5-73ee-46a9-aaa4-3040120de02b.png">

- i think for Android screen headers, it is conventional to have the left item as an icon and have the header title on the left rather than in the center. should i make this adjustment here too? instead of "close", i could use a cross icon for android. and the title would be next to the cross icon like [this ](https://developer.android.com/guide/navigation/navigation-ui) - curious for your thoughts @nityarao , it'd be cool to build a "native" feeling screen but i also see this is not the pattern we use for other screens so i'll go with what you decide

Reply:
update: sorry i missed the updated designs, i have now moved the refresh icon also to the bottom nav bar. i've been doing some research and am pretty satisfied that having this bottom nav bar with all the actions isn't a strange pattern on android :) for me, having the refresh icon there makes a big difference in making the nav bar feel more useful. so i'd like to leave it like this, and get some feedback through our internal testers on the nightly builds. i'd be happy to tweak anything that comes out of the feedback! (have updated the PR description snapshot to include the change to the bottom nav bar)